src/bin/nydusd/fs_cache.rs

```diff
@@ -773,38 +783,31 @@ impl FsCacheHandler {
             let volume_key = &file_name[1..];
             let (cookie_dir, cookie_name) = self.generate_cookie_path(&path, volume_key, &blob_id);
             let cookie_path = cookie_dir.join(&cookie_name);
-            if cookie_path.is_file() {
-                match self.inuse(&cookie_dir, &cookie_name) {
-                    Err(e) => {
-                        warn!(
-                            "blob {} call inuse err {}, cull failed!",
-                            cookie_path.display(),
-                            e
-                        );
-                        res = false;
-                    }
-                    Ok(true) => {
-                        warn!("blob {} in use, cull failed!", cookie_path.display());
+            if !cookie_path.is_file() {
+                continue;
+            }
+            let cookie_path = cookie_path.display();
+
+            match self.inuse(&cookie_dir, &cookie_name) {
+                Err(e) => {
+                    warn!("blob {} call inuse err {}, cull failed!", cookie_path, e);
+                    res = false;
+                }
+                Ok(true) => debug!("blob {} in use, skip!", cookie_path),
```

Comment:
Even if something goes unexpectedly, it would be better to keep on reclaiming other blob files. Otherwise a dougling blob file may block the gc for ever.

Reply:
Yes, so maybe we can output a warning message, set res=false , and go to next loop. Finally return error to snapshotter make it aware something going wrong?

Reply:
Fixed